xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2399,9 +2409,27 @@ public String getSkinFile(String filename, XWikiContext context)
         return getSkinFile(filename, false, context);
     }
 
-    public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext context)
+    /**
+     * Build and return a skin file url based on the given parameters.
+     * @param filename the file name of the skin file wanted
+     * @param forceSkinAction if true force the usage of directory /skins/ in the URL
+     * @param context current context for the request
+     * @param queryParameters parameters to add to the URL
+     * @return a resource URL for the asked filename
+     * @since 11.1RC1
+     */
+    @Unstable
+    public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext context,
+        Map<String, String> queryParameters)
     {
         XWikiURLFactory urlf = context.getURLFactory();
+        Map<String,String> queryParametersMap;
```

Comment:
Does not seems to be used.

Reply:
Good catch! I indeed forgot to fill it, and to use it. I'll fix that.